## Loading libraries and dataset

In [1]:
import mlxtend
import pandas as pd

from mlxtend.preprocessing import TransactionEncoder
from mlxtend.frequent_patterns import apriori, association_rules 

dataset = [['Skirt', 'Sneakers', 'Scarf', 'Pants', 'Hat'],
           ['Sunglasses', 'Skirt', 'Sneakers', 'Pants', 'Hat'],
           ['Dress', 'Sandals', 'Scarf', 'Pants', 'Heels'],
           ['Dress', 'Necklace', 'Earrings', 'Scarf', 'Hat', 'Heels', 'Hat'],
           ['Earrings', 'Skirt', 'Skirt', 'Scarf', 'Shirt', 'Pants']]



## Transform dataset into a one-hot-encoded Dataframe.

In [2]:
te=TransactionEncoder()
te_ary=te.fit(dataset).transform(dataset)    #Apply one-hot-encoding on our dataset
df=pd.DataFrame(te_ary, columns=te.columns_)  #Creating a new DataFrame from our Numpy array
df

Dress  Earrings    Hat  Heels  Necklace  Pants  Sandals  Scarf  Shirt  \
0  False     False   True  False     False   True    False   True  False   
1  False     False   True  False     False   True    False  False  False   
2   True     False  False   True     False   True     True   True  False   
3   True      True   True   True      True  False    False   True  False   
4  False      True  False  False     False   True    False   True   True   

   Skirt  Sneakers  Sunglasses  
0   True      True       False  
1   True      True        True  
2  False     False       False  
3  False     False       False  
4   True     False       False

## Support:

In [3]:
#Support calculation.
    
frequent_itemsets=apriori(df, min_support=0.5, use_colnames=True)
frequent_itemsets

support        itemsets
0      0.6           (Hat)
1      0.8         (Pants)
2      0.8         (Scarf)
3      0.6         (Skirt)
4      0.6  (Scarf, Pants)
5      0.6  (Skirt, Pants)

Le tableau si dessus montre les articles les plus fréquemment achetés, le stock manager devrait se réapprovisionner régulièrement pour éviter la rupture de stock.

## Confidence:

In [4]:
# Confidence calculation - associate itemsets with confidence over 60%.

association_rules(frequent_itemsets,metric="confidence",min_threshold=0.6)

antecedents consequents  antecedent support  consequent support  support  \
0     (Scarf)     (Pants)                 0.8                 0.8      0.6   
1     (Pants)     (Scarf)                 0.8                 0.8      0.6   
2     (Skirt)     (Pants)                 0.6                 0.8      0.6   
3     (Pants)     (Skirt)                 0.8                 0.6      0.6   

   confidence    lift  leverage  conviction  
0        0.75  0.9375     -0.04         0.8  
1        0.75  0.9375     -0.04         0.8  
2        1.00  1.2500      0.12         inf  
3        0.75  1.2500      0.12         1.6

Selon le calcule de la confidence, on peux déduire les gégles suivantes:
- aprés avoir acheté un pants, le client à 75% de chance d'acheter un Scarf ou une Skirt.
- aprés avoir acheté un Scarf, le client à 75% de chance d'acheter un Pants.
- aprés avoir acheté une Skirt, le client à 100% de chance d'acheter un Pants.

Ces articles devront étre positionnés cote à cote.

## Lift :

In [5]:
# Lift calculation.
association_rules(frequent_itemsets,metric="lift",min_threshold=1.05)

antecedents consequents  antecedent support  consequent support  support  \
0     (Skirt)     (Pants)                 0.6                 0.8      0.6   
1     (Pants)     (Skirt)                 0.8                 0.6      0.6   

   confidence  lift  leverage  conviction  
0        1.00  1.25      0.12         inf  
1        0.75  1.25      0.12         1.6

Selon le calcule du lift, Il est plus probable d’acheter un Pants et une Skirt avec un lift de 1.25 que un Pants seul ou une Skirt seule.